datalab/datalab_session/data_operations/median.py

```diff
@@ -42,63 +40,40 @@ def wizard_description():
-            memmap_paths = []
+        self.set_percent_completion(0.4)
 
-            for index, file_info in enumerate(input_files):
-                basename = file_info.get('basename', 'No basename found')
-                archive_record = get_archive_from_basename(basename)
+        stacked_data = stack_arrays(image_data_list)
 
-                try:
-                    fits_url = archive_record[0].get('url', 'No URL found')
-                except IndexError:
-                    continue
+        median = np.median(stacked_data, axis=2)
 
-                with fits.open(fits_url, use_fsspec=True) as hdu_list:
-                    data = hdu_list['SCI'].data
-                    memmap_path = os.path.join(temp_dir, f'memmap_{index}.dat')
-                    memmap_array = np.memmap(memmap_path, dtype=data.dtype, mode='w+', shape=data.shape)
-                    memmap_array[:] = data[:]
-                    memmap_paths.append(memmap_path)
+        hdu_list = create_fits(cache_key, median)
 
-                self.set_percent_completion(index / file_count)
+        # Create the output files to be stored in S3
+        fits_path           = tempfile.NamedTemporaryFile(suffix=f'{cache_key}.fits').name
+        large_jpg_path      = tempfile.NamedTemporaryFile(suffix=f'{cache_key}-large.jpg').name
+        thumbnail_jpg_path  = tempfile.NamedTemporaryFile(suffix=f'{cache_key}-small.jpg').name
+        
+        hdu_list.writeto(fits_path)
+        fits_to_jpg(fits_path, large_jpg_path, width=median.shape[0], height=median.shape[1])
+        fits_to_jpg(fits_path, thumbnail_jpg_path)
 
-            image_data_list = [
-                np.memmap(path, dtype=np.float32, mode='r', shape=memmap_array.shape)
-                for path in memmap_paths
-            ]
+        self.set_percent_completion(0.7)
 
-            # Crop fits image data to be the same shape then stack
-            min_shape = min(arr.shape for arr in image_data_list)
-            cropped_data_list = [arr[:min_shape[0], :min_shape[1]] for arr in image_data_list]
-            stacked_data = np.stack(cropped_data_list, axis=2)
+        # Save Fits and Thumbnails in S3 Buckets
+        fits_url            = add_file_to_bucket(f'{cache_key}/{cache_key}.fits', fits_path)
+        large_jpg_url       = add_file_to_bucket(f'{cache_key}/{cache_key}-large.jpg', large_jpg_path)
+        thumbnail_jpg_url   = add_file_to_bucket(f'{cache_key}/{cache_key}-small.jpg', thumbnail_jpg_path)
 
-            # Calculate a Median along the z axis
-            median = np.median(stacked_data, axis=2)
+        self.set_percent_completion(0.9)
 
-            cache_key = self.generate_cache_key()
-            header = fits.Header([('KEY', cache_key)])
-            primary_hdu = fits.PrimaryHDU(header=header)
-            image_hdu = fits.ImageHDU(median)
-            hdu_list = fits.HDUList([primary_hdu, image_hdu])
+        output = {'output_files': [large_jpg_url, thumbnail_jpg_url]}
```

Comment:
The output should still just be one image, but it is an object/dict, so it can contain multiple keys. Right now I've had the outputs mirroring the inputs, i.e. 
`{'source': 'archive', 'basename': 'asldfkljasldkfj'}`. For your output I would like to add two keys for the thumbnail URLs, so it should look like this:
`{'output_files': [{'source': 'datalab', 'basename': 'cache_key', 'small_thumbnail_url': url, 'large_thumbnail_url': url}]}`